pkg/operator/backupcontroller/backupcontroller.go

```diff
@@ -0,0 +1,341 @@
+		return strings.Compare(backupsToRun[i].Name, backupsToRun[j].Name) < 0
+	})
+
+	for _, backup := range backupsToRun {
+		err := createBackupJob(ctx, backup, c.targetImagePullSpec, jobsClient, backupsClient)
+		if err != nil {
+			return err
+		}
+
+		// only ever reconcile one item at a time
+		// TODO(thomas): we should mark the remainder as "Skipped"
+		return nil
+	}
+
+	return nil
+}
+
+func indexJobsByBackupLabelName(jobs *batchv1.JobList) map[string]batchv1.Job {
+	m := map[string]batchv1.Job{}
+	if jobs == nil {
+		return m
+	}
+
+	for _, j := range jobs.Items {
+		backupCrdName := j.Labels[backupJobLabel]
+		if backupCrdName != "" {
+			m[backupCrdName] = *j.DeepCopy()
+		}
+	}
+
+	return m
+}
+
+func findRunningJobs(jobs *batchv1.JobList) []batchv1.Job {
+	var running []batchv1.Job
+	if jobs == nil {
+		return running
+	}
+
+	for _, j := range jobs.Items {
+		if !isJobFinished(&j) {
+			running = append(running, *j.DeepCopy())
+		}
+	}
+
+	return running
+}
+
+// isJobFinished checks whether the given Job has finished execution.
+// It does not discriminate between successful and failed terminations.
+func isJobFinished(j *batchv1.Job) bool {
+	for _, c := range j.Status.Conditions {
+		if (c.Type == batchv1.JobComplete || c.Type == batchv1.JobFailed) && c.Status == corev1.ConditionTrue {
+			return true
+		}
+	}
+	return false
+}
+
+func reconcileJobStatus(ctx context.Context,
+	jobClient batchv1client.JobInterface,
+	backupClient backupv1client.EtcdBackupInterface,
+	job batchv1.Job,
+	backup backupv1alpha1.EtcdBackup) error {
+
+	jobState := batchv1.JobConditionType("")
+	for _, c := range job.Status.Conditions {
+		// the types and type transitions are compatible between jobs and our backup states
+		if (c.Type == batchv1.JobComplete || c.Type == batchv1.JobFailed) && c.Status == corev1.ConditionTrue {
+			jobState = c.Type
+			break
+		}
+	}
+
+	// we only reconcile completed/failed jobs
+	if string(jobState) == "" {
+		return nil
+	}
+
+	if backup.Labels == nil {
+		backup.Labels = map[string]string{}
+	}
+
+	backup.Labels["state"] = "processed"
+
+	// we're updating the backup and its status first, in case of a failure we can still reconcile the job without it
+	_, err := backupClient.Update(ctx, &backup, v1.UpdateOptions{})
+	if err != nil {
+		return fmt.Errorf("error while updating backup [%s]: %w", backup.Name, err)
+	}
+
+	// update above has changed the CRD, to avoid conflicts we grab the latest version from the API
+	bp, err := backupClient.Get(ctx, backup.Name, v1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error while getting backup for updating status [%s]: %w", backup.Name, err)
+	}
+
+	bp.Status.Conditions = append(bp.Status.Conditions,
+		v1.Condition{
+			Type:               "BackupCompleted",
+			Reason:             "BackupCompleted",
+			Message:            fmt.Sprintf("%s", jobState),
+			Status:             v1.ConditionTrue,
+			LastTransitionTime: metav1.NewTime(time.Now()),
+		})
```

Comment:
We do have `BackupCompleted` and `BackupFailed` as conditions for the EtcdBackup CR.
https://github.com/openshift/api/blob/master/operator/v1alpha1/types_etcdbackup.go#L81-L85

So perhaps we can map `batchv1.JobFailed` to `BackupFailed` on the EtcdBackup status.

Reply:
perfect, let me use that.